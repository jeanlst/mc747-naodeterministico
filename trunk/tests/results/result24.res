Semantic Error
Redeclaracao da funcao print nao eh possivel
Redeclaracao da funcao print nao eh possivel
As funções reads e readln aceitam apenas variáveis como parâmetro
Variável não pode ser passada por referência