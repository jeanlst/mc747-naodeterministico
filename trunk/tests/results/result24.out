java.lang.ClassCastException: pl747.tabelaSimbolos.ConstSymb cannot be cast to pl747.tabelaSimbolos.VarSymb
	at pl747.semantico.VarOp.check(VarOp.java:40)
	at pl747.semantico.DiadOp.check(DiadOp.java:416)
	at pl747.semantico.ForStat.check(ForStat.java:90)
	at pl747.semantico.CompoundStat.check(CompoundStat.java:49)
	at pl747.semantico.FunctionDeclaration.check(FunctionDeclaration.java:164)
	at pl747.semantico.Program.check(Program.java:195)
	at CompilerFacade.compile(CompilerFacade.java:48)
	at Compiler.main(Compiler.java:71)
java.lang.ClassCastException: pl747.tabelaSimbolos.ConstSymb cannot be cast to pl747.tabelaSimbolos.VarSymb
	at pl747.semantico.VarOp.check(VarOp.java:40)
	at pl747.semantico.DiadOp.check(DiadOp.java:416)
	at pl747.semantico.ForStat.check(ForStat.java:90)
	at pl747.semantico.CompoundStat.check(CompoundStat.java:49)
	at pl747.semantico.FunctionDeclaration.check(FunctionDeclaration.java:164)
	at pl747.semantico.Program.check(Program.java:195)
	at CompilerFacade.compile(CompilerFacade.java:48)
	at Compiler.main(Compiler.java:71)
java.lang.ClassCastException: pl747.tabelaSimbolos.ConstSymb cannot be cast to pl747.tabelaSimbolos.VarSymb
	at pl747.semantico.VarOp.check(VarOp.java:40)
	at pl747.semantico.DiadOp.check(DiadOp.java:416)
	at pl747.semantico.ForStat.check(ForStat.java:90)
	at pl747.semantico.CompoundStat.check(CompoundStat.java:49)
	at pl747.semantico.DoStat.check(DoStat.java:59)
	at pl747.semantico.CompoundStat.check(CompoundStat.java:49)
	at pl747.semantico.FunctionDeclaration.check(FunctionDeclaration.java:164)
	at pl747.semantico.Program.check(Program.java:195)
	at CompilerFacade.compile(CompilerFacade.java:48)
	at Compiler.main(Compiler.java:71)
java.lang.ClassCastException: pl747.tabelaSimbolos.ConstSymb cannot be cast to pl747.tabelaSimbolos.VarSymb
	at pl747.semantico.VarOp.check(VarOp.java:40)
	at pl747.semantico.DiadOp.check(DiadOp.java:416)
	at pl747.semantico.ForStat.check(ForStat.java:90)
	at pl747.semantico.CompoundStat.check(CompoundStat.java:49)
	at pl747.semantico.FunctionDeclaration.check(FunctionDeclaration.java:164)
	at pl747.semantico.Program.check(Program.java:195)
	at CompilerFacade.compile(CompilerFacade.java:48)
	at Compiler.main(Compiler.java:71)
java.lang.ClassCastException: pl747.tabelaSimbolos.ConstSymb cannot be cast to pl747.tabelaSimbolos.VarSymb
	at pl747.semantico.VarOp.check(VarOp.java:40)
	at pl747.semantico.DiadOp.check(DiadOp.java:416)
	at pl747.semantico.ForStat.check(ForStat.java:90)
	at pl747.semantico.CompoundStat.check(CompoundStat.java:49)
	at pl747.semantico.FunctionDeclaration.check(FunctionDeclaration.java:164)
	at pl747.semantico.Program.check(Program.java:195)
	at CompilerFacade.compile(CompilerFacade.java:48)
	at Compiler.main(Compiler.java:71)
